components/wso2is.notification.event.handlers/src/main/java/org/wso2/is/notification/internal/NotificationServiceComponent.java

```diff
@@ -32,6 +33,8 @@ protected void activate(ComponentContext componentContext) throws Exception {
         BundleContext bundleContext = componentContext.getBundleContext();
         serviceRegistration =
                 bundleContext.registerService(OAuthEventInterceptor.class, new ApimOauthEventInterceptor(), null);
+        serviceRegistration =
```

Comment:
shall we use a new variable and unregister at osgi deactivate

Reply:
We already unregister the `serviceRegistration` in [1]. Do we still need to introduce new variable?

[1] https://github.com/wso2-extensions/apim-km-wso2is/blob/master/components/wso2is.notification.event.handlers/src/main/java/org/wso2/is/notification/internal/NotificationServiceComponent.java#L90